src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/DisposeObjectsBeforeLosingScopeTests.cs

```diff
@@ -2128,6 +2128,64 @@ End Function
             }.RunAsync();
         }
 
+        [Fact, WorkItem(3305, "https://github.com/dotnet/roslyn-analyzers/issues/3305")]
+        public async Task LocalWithRefStructDisposableAssignment_NotDisposed_Diagnostic()
+        {
+            await new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
+                TestCode = @"
+using System;
+
+ref struct RefStructDisposable
+{
+    public void Dispose()
```

Comment:
Is it possible to have `DisposeAsync` by convention?

Reply:
Yes and no.

You can declare the method, and there would be no `using`-related error in `await using var disposable = new RefLikeAsyncDisposable()`.

But there will be another error: at this moment ref-like types are completely prohibited in async methods. So `var disposable` will lead to `CS4012`: Parameters or locals of type 'RefLikeAsyncDisposable' cannot be declared in async methods or lambda expressions.

Reply:
Might be worth adding a test for this case with an expected error which could be fixed in the future, WDYT?

Reply:
I don't understand what kind of test do you suggest exactly.

Due to the `CS4012` I don't think this PR needs to be concerned of `async` for now.

Reply:
I was thinking about something like (code was written directly so might have some errors):

```csharp
await new VerifyCS.Test
{
	ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithAsyncInterfaces,
	TestCode = @"
using System;
ref struct RefStructDisposableAsync
{
    public ValueTask DisposeAsync()
    {
    }
}
class Test
{
    public static async Task M1()
    {
        await using var e = new RefStructDisposableAsync();
    }
}
",
		ExpectedDiagnostics =
		{
			VerifyCS.CompilerError("CS4012").WithLocation(),
		}
	}.RunAsync();
}
```

Reply:
I think it should be fine without tests for `Async` variants for now. If it is ever allowed in the future code style and semantics might change a bit.